index.html

```diff
@@ -89,48 +97,142 @@
+      .nice();    
+   
     // AXES
-
+    var xAxis = d3.svg.axis()
+      .scale(x)
+      .tickSize(-height)
+      .orient("bottom");
+    var yAxis = d3.svg.axis()
+      .scale(y)
+      .tickSize(-width)
+      .orient("left");
+    svg.append("g")
+      .classed("x axis", true)
+      .attr("transform", "translate(0," + (height-margin.top) + ")")
+      .call(xAxis);
+    svg.append("g")
+      .classed("y axis", true)
+      .attr("transform","translate(" + margin.left + ",0)")
+      .call(yAxis);
+    var line = d3.svg.line()
+      .x(function(d) { return xScale(d.x); })
+      .y(function(d) { return yScale(d.y); });
+    svg.append("text")
+        .attr("text-anchor", "middle")
+        .attr("transform", "translate(0,"+(height/2)+")rotate(-90)")
+        .text("District Partisanship");
+    svg.append("text")
+        .attr("text-anchor", "middle")
+        .attr("transform", "translate("+ (width/2) +","+(height + margin.top + (margin.bottom/2))+")")
+        .text("Candidate CF Score");
     // EVENT HANDLERS
-
-    d3.select('#year-range').on('change', function () {
-      options.yearRange = d3.event.target.value;
+    d3.select('#party').on('change', function () {
+      options.party = d3.event.target.value;
       update();
     });
   }
-
   function update() {
     // TRANSFORM
     txData = data.slice();
-
-    if (options.yearRange !== 'all') {
-      var yearRange = options.yearRange.split('-');
+    if (options.party !== 'all') {
+      var party = options.party;
       txData = txData.filter(function (d) {
-        return d.year >= yearRange[0] && d.year <= yearRange[1];
+        return d.party === party;
       });
     }
-
     // SCATTERPLOT
-
+    var points = svg.selectAll('.point')
+      .data(txData);
+
+    points.enter().append('circle')
+      .classed("point", true)
+      .attr('r', 5);
+    
+    points
+      .classed('point',true)
+      .classed("R", function (d) {return d.party === "rep";})
+      .classed("D", function (d) {return d.party === "dem";})
+      .attr('cx', function (d){return x(d.cfScore) ;})
+      .attr('cy', function (d){return y(d.distPart) ;});
+
+    points.exit().remove();
+    
     // HISTOGRAMS
+    // X-HIST
+ 
+    var xdata = data.map(function(d) { return d.cfScore; });
+    var dataHistX = d3.layout.histogram()
+    .bins(x.ticks())
+    (xdata);
+
+    var ymax = d3.max(dataHistX, function (i) {return i.y});
+
+    var vert = d3.scale.linear()
+      .domain( [0, ymax] )
+      .range([ y.ticks()[0] , y.ticks()[1]]);
+
+    var bar = svg.selectAll(".bar")
+        .data(dataHistX)
+      .enter().append("g")
+        .attr("class", "bar");
+        // .attr("transform","translate(0," + margin.top + ")");
+
+    bar.append("rect")
+      .attr("x", function (d) { return x(d.x); })
+      .attr("y", function (d) { return y(vert(d.y)); })
+      .attr("width", width/(x.ticks().length ))
+      .attr("height", function (d) { return d.y; });
+
```

Comment:
This is on the right track. There's no reason you couldn't do `[y(y.ticks()[1]) , y(y.ticks()[0])]`, but also I see two other potential approaches:

* You could use `Math.abs()` to get the absolute value.
* You could use the `dataHistY[0].dx`, which already has it calculated for you.

Also note that on line 234 and 267, we don't need to worry about margins here. They've already been subtracted from the height and width available to us, and we transformed the group that everything's inside to offset everything by those margins.